webpack.config.js

```diff
@@ -0,0 +1,78 @@
+const path = require("path");
+const { CleanWebpackPlugin } = require("clean-webpack-plugin");
+const HTMLWebpuckPlugin = require("html-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+
+const isProd = process.env.NODE_ENV === "production";
+const isDev = !isProd;
+
+const filename = (ext) => {
+  return isDev ? `bundle.${ext}` : `bundle.[hash].${ext}`;
+};
+
+module.exports = {
+  context: path.resolve(__dirname, "src"),
+  mode: "development",
```

Comment:
there is no reason to set production --mode in scripts, determine it here
`mode: isDev ? 'development' : 'production';`